lib/active_record/connection_adapters/crate_adapter.rb

```diff
@@ -125,13 +124,26 @@ def supports_migrations?
         true
       end
 
+      def supports_datetime_with_precision?
+        true
+      end
+
       def connect
         @connection = CrateRuby::Client.new(["#{@host}:#{@port}"])
+
+        # Monkeypatch to make the client instance provide `supports_datetime_with_precision`.
+        # FIXME: Implement within `CrateRuby::Client`?
+        @adapter_supports_datetime_with_precision = supports_datetime_with_precision?
+        def @connection.supports_datetime_with_precision?
+          return @adapter_supports_datetime_with_precision
+        end
+
       end
 
       def columns(table_name) #:nodoc:
         cols = @connection.table_structure(table_name).map do |field|
-          name = dotted_name(field[2])
+          # TODO: Review. What if the table structure changes again?
+          name = dotted_name(field[12])
           CrateColumn.new(name, nil, field[4], nil)
         end
         cols
```

Comment:
This feels a bit fishy, and I don't even know if it is correct.

Reply:
For inquiring the table columns, something along the lines of [sqlalchemy/dialect.py#L251-L263](https://github.com/crate/crate-python/blob/0.27.2/src/crate/client/sqlalchemy/dialect.py#L251-L263) should be used, no?